### DISCUSSION TITLE
[vcpkg] vcpkg requires dumpbin.exe with gcc on windows

### DISCUSSION THREAD

**Discussion by vadi2** [microsoft/vcpkg#17868] [Ideas]
**Describe the bug**
Very similar to https://github.com/microsoft/vcpkg/issues/11467, but for GCC - `objdump` is not used an an error is given: `       | Neither dumpbin nor llvm-objdump could be found. Can not take care of dll dependencies.`

**Environment**
- OS: Windows
- Compiler: MinGW 8.1.0

**To Reproduce**
Build with vcpkg and mingw.

**Expected behavior**
Deploy to work with `objdump` like it already works with `llvm-objdump`.

**Failure logs** 
-(please attached failure logs)

**Additional context**
https://github.com/microsoft/vcpkg/pull/11898 already has a solution. The same solution might work for `objdump`.

**Comment by vadi2** [comment]
Workaround for the time being is to use `-DVCPKG_APPLOCAL_DEPS=OFF`.

**Comment by vadi2** [comment]
dumpbin is not needed with mingw, it has its own objdump.

**Comment by vadi2** [comment]
I don't have dumpbin. That's why it's not found.

**Comment by vadi2** [comment]
I'm not mixing VS with MinGW. I'm afraid it is you who is confused. Thanks for the help anyway!

**Comment by strega-nil** [comment]
@vadi2 If you would like, you can look into adding this feature; unfortunately, we do not have experience on the team with MinGW's stuff.

@sfhacker Please keep your comments on-topic.

**Comment by vadi2** [comment]
@longnguyen2004 are you able to look into this one by chance?

**Comment by longnguyen2004** [comment]
@vadi2 It's related to the [applocal.ps1](https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/msbuild/applocal.ps1) script, which copies the dlls over to the build directory automatically upon building. For now it only supports MS dumpbin and LLVM llvm-objdump to find the dependencies. I can try to add support for GCC objdump.
For now, please disable that feature by setting `VCPKG_APPLOCAL_DEPS` to `OFF`.

@sfhacker Please refrain from commenting if you don't understand vcpkg internals.

**Comment by longnguyen2004** [comment]
@dg0yt Maybe look into this one? I'm busy with the 2 other PRs

**Reply by dg0yt** [reply]
@longnguyen2004 I think I will need to finish the GDAL/Linux backlog first. That's more than 2 PRs. And I even don't use this app-local mode at the moment.

@vadi2 Perhaps you want try a modifcation yourself? `objdump` output seems to be quite similar to `llvm-objdump`, except that the lines of interest start with \<TAB>, not four spaces. 

~~~
objdump -p file.exe | ? { $_ -match "^\tDLL Name: .*\.dll" } | % { $_ -replace "^\tDLL Name: ","" }
libmagic.dll
VCRUNTIME140.dll
api-ms-win-crt-heap-l1-1-0.dll
api-ms-win-crt-runtime-l1-1-0.dll
api-ms-win-crt-convert-l1-1-0.dll
api-ms-win-crt-environment-l1-1-0.dll
api-ms-win-crt-string-l1-1-0.dll
api-ms-win-crt-stdio-l1-1-0.dll
api-ms-win-crt-locale-l1-1-0.dll
api-ms-win-crt-math-l1-1-0.dll
KERNEL32.dll
~~~

So look at the https://github.com/microsoft/vcpkg/pull/11898/files, clone the two lines for `llvm-objdump`, and adjust the pattern as above.

(Personal note: CMake has its own cross-platform abstraction for resolution of runtime dependencies.)

**Reply by dg0yt** [reply]
Pull request: https://github.com/microsoft/vcpkg/pull/18027
